app/models/team.rb

```diff
@@ -18,6 +18,37 @@ def to_s
     name
   end
 
+  def user_deletion_history
+    PaperTrail::Version.
+      where(event: 'destroy', item_type: 'UserTeam').
+      where("object ->>'group_id' = '?'", id)
+  end
+
+  def user_deletion_report_item(record)
+    {
+      user_id: record[0]['user_id'],
+      user_email: User.find_by(id: record[0]['user_id'])&.email,
+      team_id: record[0]['group_id'],
+      team_name: Team.find_by(id: record[0]['group_id'])&.name,
+      removed_at: record[1],
+      whodunnit_id: record[2],
+      whodunnit_email: User.find_by(id: record[2])&.email,
+    }
+  end
+
+  def user_deletion_history_report(email = nil)
+    user_deletion_history.
+      order(created_at: :desc).
+      limit(5000).
+      pluck(:object, :created_at, :whodunnit).
+      select { |record|
+        email.nil? || User.find_by(id: record[0]['user_id'])&.email == email
+      }.
```

Comment:
**suggestion:** You can probably use argument destructuring here as well to keep this more readable:

```suggestion
      select { |object, _, _|
        email.nil? || User.find_by(id: object['user_id'])&.email == email
      }.
```